gitbook-docs/data_model.md

```diff
@@ -1,104 +1,178 @@
+        "GP": {
+          "talker": "GP",
+          "sentences": {
+            "RMC": "2017-04-03T06:14:04.451Z"
+          },
+          "II": {
+            "talker": "II",
+            "sentences": {
+              "HDM": "2017-05-16T05:15:54.006Z"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+There are several top level attributes or keys which are always present and others which are optional. The `version`
+key specifies which version of the Signal K specification is being used and must always present in a full Signal K
+model. Also always present in the full model is the `self` key. The value of `self` is the key within the `vessels`
+object which is the local boat. Effectively, it is a pointer into the `vessels` object.
+
+Below the `vessels` object is a list of vessels, identified by their MMSI number or a generated unique ID. There may be
+many vessels if data has been received from AIS or other sources. The format for each vessel’s data uses the same
+standard Signal K structure but may not have the same content; likely you will not have as much data about other
+vessels as you have about your own.
+
+At the same level as `vessels` is `sources`. This contains a list of sources the data was obtained from. Each data
+object within a `vessel` may have a `$source` key which point to a source withing `sources`. Several data objects may
+reference the same `source` since a single NMEA sentence or PGN may map to multiple keys in Signal K.
+
+Alternatively the source data may be embedded directly in place of the `$source` by using the `source` key:
 
 ```json
 {
   "vessels": {
-    "urn:mrn:signalk:uuid:c0d79334-4e25-4245-8892-54e8ccc8021d": {
-      "version": "0.1",
-      "name": "motu",
-      "mmsi": "2345678",
-      "source": "self",
-      "timezone": "NZDT",
+    "urn:mrn:signalk:uuid:705f5f1a-efaf-44aa-9cb8-a0fd6305567c": {
       "navigation": {
-        "state": {
-          "value": "sailing",
-          "source": "self",
-          "timestamp": "2014-03-24T00:15:41Z"
-        },
-        "headingTrue": {
-          "value": 2.3114,
-          "$source": "nmea0183-1.II",
-          "sentence": "HDT",
-          "timestamp": "2014-03-24T00:15:41Z"
-        },
-        "speedThroughWater": {
-          "value": 2.556,
-          "$source": "n2k-1.160",
-          "pgn": 128259,
-          "timestamp": "2014-03-24T00:15:41Z"
-        },
         "position": {
-          "longitude": 23.53885,
-          "latitude": 60.0844,
-          "$source": "nmea0183-2.GP",
-          "timestamp": "2014-03-24T00:15:42Z",
-          "sentence": "GLL"
+          "value": {
+            "altitude": 0.0,
+            "latitude": 37.81479,
+            "longitude": -122.44880152
+          },
+          "source": {
+            "label": "ttyUSB0",
+            "type": "NMEA0183",
+            "talker": "GP",
+            "sentence": "$GPRMC,061404.000,A,4117.6201,S,17314.8224,E,0.38,354.82,030417,,*11"
```

Comment:
`Sentence` is in the schema and a string, but schema documentatio says it is the _sentence type_, not the sentence data. Node server handles only the three letter sentence type, not data.

If we define a way to pass along the raw data then we should do so explicitly and cater for N2K data as well. 

I think it would be a useful feature for debugging and testing and handling propietary N2K pgns, but that warrants a separate issue that includes software changes in RI.